backend/backend-standard-and-best-practices.md

```diff
@@ -0,0 +1,278 @@
+
+You can read more about this and other status codes in [this link](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).
+
+## 6- API Documentation
+
+There are lots of different tools in the market to document APIs. Here in Wolox we decided to use Openapi 3 (A.K.A Swagger) as a standard documentation tool in all of our techs. The reasons behind this decision were some of the advantages OpenApi prodvides:
+
+* **Quick to use**: The properties that you need to fill in every endpoint are always the same and there are only a few. You can also frequently reuse components and the generic information is simple to provide.
+* **Wide range of available features**: From describing parameters, making authentication and authorization schemes, to even using the generated documentation of an endpoint to mock test the endpoint in question.
+* **Intuitive and easy to use interactive UI**: Swagger uses a GUI to show the user how to make requests to the API and what responses will look like. 
+* **Maintainable and easily extendable**: Swagger allows using constants files from your API. 
+* **Scan and auto-generate**: For typed languages, Swagger can scan and auto-generate documentation.  
+
+You can read more a about Swagger in this [this link](https://swagger.io/docs/).
+
+### 7- Routes
+
+While it's impossible to define every single route name possibility, we chose a few examples to act as guidelines
+
+#### Wrong
+* *`GET /get_users`, `GET /get_user/:user_id`, `GET /get_user?user_id=1`, `GET /get?model=user&user_id=1`*: while we don't need to be 100% RESTful, we should use the basic idea of resources and verbs as separate things.
+* *`GET /user_summary/1`*: use nested resources for cases where a resource is deeply linked to another resource.
+* *`POST /create_user`*: a POST to a route named `create_X` is redundant, just make a POST to `/users/`
+* *`DELETE /delete_user `*: likewise, a DELETE to a route named `delete_X` is redudant.
+* *`POST /users/update`*: don't use post for an update. Use either patch or put depending on whether you need to pass the whole instance or just the updated fields. Use POST for more generic endpoints that escape the common RESTful resource logic though.
+* *`GET /users/update?field1=a&field2=b&...`*: don't use a GET for an update with query params for fields.
+* *Spanglish*: or any other combination of non-english terms unless it's strictly necessary.
+* *Camel case*: the standard for Wolox backends is to use `lower_snake_case` for routes.
+* *Restricting yourself to database models*: REST resources don't need to have a 1:1 relationship with database models. If it makes semantic sense to break one model up into multiple resources or to join multiple models using an abstract model, that's correct.
+
+#### Right
+* *`GET /users`*
+* *`GET /users/:user_id`*
+* *`PATCH /users/:user_id` and `PUT /users/:user_id`*
+* *`POST /users`*
+* *`DELETE /users/:user_id`*
```

Comment:
I think `:user_id` should be replace for `id`. Usually in REST routes the path define the resources (e.g.: `/users/`) so after that we can use directly `:id` as path param